File: 0060-permutation-sequence/0060-permutation-sequence.cpp
class Solution {
public:
    string getPermutation(int n, int k) {
        vector<int> nums;
        int fact = 1;
        for(int i = 1; i <= n; i++){
            nums.push_back(i);
            fact *= i;
        }

        fact /= n;
        k = k - 1; // 0 based indexing
        string ans = "";

        while(true){
            ans.push_back(nums[k/fact] + '0');
            nums.erase(nums.begin() + (k/fact));

            if(nums.size() == 0)
                break;
            
            k = k % fact;
            fact = fact / nums.size();          
        }

        return ans;
    }
};

File: 0060-permutation-sequence/0060-permutation-sequence.py
class Solution:
    def getPermutation(self, n: int, k: int) -> str:
        nums = list(range(1,n+1))
        fact = 1
        for i in range(1,n):
            fact *= i
        
        k -= 1
        ans = ""

        while True:
            ans += str(nums[k//fact])
            nums.pop(k//fact)

            if len(nums) == 0:
                break
            
            k = k % fact
            fact = fact // len(nums)
        

        return ans
        